lib/receipt.js

```diff
@@ -0,0 +1,210 @@
+'use strict';
+
+/**
+ * @module striim-sdk
+ */
+
+const {hashObject, sign, ethHash, isSignedBy} = require('./utils');
+const Payment = require('./payment');
+const MonetaryAmount = require('./monetary-amount');
+
+const _provider = new WeakMap();
+const _payment = new WeakMap();
+const _hash = new WeakMap();
+const _signature = new WeakMap();
+const _nonce = new WeakMap();
+const _senderNonce = new WeakMap();
+const _recipientNonce = new WeakMap();
+const _senderCurrentBalance = new WeakMap();
+const _senderPreviousBalance = new WeakMap();
+const _senderSingleFee = new WeakMap();
+const _senderNetFee = new WeakMap();
+const _recipientCurrentBalance = new WeakMap();
+const _recipientPreviousBalance = new WeakMap();
+const _recipientNetFees = new WeakMap();
+const _netTransfer = new WeakMap();
+const _singleTransfer = new WeakMap();
+
+const PAYMENT_RECEIPT_HASHED_PROPERTIES = [
+    'seals.wallet.signature.v',
+    'seals.wallet.signature.r',
+    'seals.wallet.signature.s',
+    'nonce',
+    'sender.nonce',
```

Comment:
They are indirectly included because they are part of the hash of the underlying Payment and since the payment signature (seals.wallet.signature.*) is included, there shouldnt be a need to re-hash those.